src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -456,8 +455,8 @@ private void AdvanceReader(BufferSegment consumedSegment, int consumedIndex, Buf
 
                     _length -= examinedBytes;
 
-                    _lastExamined = examinedSegment;
-                    _lastExaminedIndex = examinedIndex;
+                    // Store the absolute position
+                    _lastExaminedIndex = examinedSegment.RunningIndex + examinedIndex;
```

Comment:
Add a test for scenario you've fixed with this.

Reply:
Nothing was fixed here. It was just an implementation change from storing 2 things to storing one.

Reply:
If I understand correctly you've fixed the problem where _lastExamined might be returned to the pool and reused by not storing it at all.